### ISSUE TITLE
lengths mismatch error  when using MixCut in icefall training

### ISSUE THREAD

**Issue by zac-wang-nv** [lhotse-speech/lhotse#751]
Hi everyone,

I am using external data mixed with noise for training, and try to exploit the nice feature from lhotse that I don't need to actually mix audio but just to mix the manifests from those two different dataset(base and noise). When loading the MixCut during training, I noticed a mismatch error in lengths and it would cause the training to fail. 

If the noise cut is longer than the base cut, which is a common setup I think, then the input's features num_frames would be longer than its supervision's num_frames. however the training code seems to assume those two lengths(feature length and supervision length) are identical, for instance, [`x_lens` argument from icefall's conformer model](https://github.com/k2-fsa/icefall/blob/5379c8e9fa13f6f2364b4a0db89fa3074266fb58/egs/librispeech/ASR/pruned_transducer_stateless5/conformer.py#L96) comes from length of supervisions which are different from `x` , which comes from features directly. 

Currently I am hacking the supervision length to be the same with the features loaded from disk, but I think it might worth fixed in a more general and natural way.

**Comment by pzelasko** [comment]
How are you exactly creating the mixed cuts? `lhotse.dataset.CutMix` and `lhotse.CutSet.mix` automatically truncate the result of mixing to match the duration of the original (speech) cut. If you are mixing cuts manually with `lhotse.Cut.mix`, you have to make sure to first truncate the noise cut to the speech cut's duration with `noise_cut = noise_cut.truncate(duration=speech_cut.duration)`.

You can follow Icefall's example with mixing in MUSAN as a reference, see:
- https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/local/compute_fbank_musan.py which cuts MUSAN into 5 second segments and extracts the features
- https://github.com/k2-fsa/icefall/blob/5379c8e9fa13f6f2364b4a0db89fa3074266fb58/egs/librispeech/ASR/tdnn_lstm_ctc/asr_datamodule.py#L224-L234 which uses a `CutMix` transform to mix MUSAN into training data (it keeps sampling 5s cuts until it mixes in the full duration of a training example)

**Comment by zzzacwork** [comment]
> 

I am using CutSet.mix, and then store the MixCut manifests to disk. 
yeah, you are right!  I read the code from CutSet.mix again and found that my code didn't do the truncate since it set the duration parameter to the max duration from that CutSet ... I probably shouldn't do it in the first place. 

Is there a way for me to truncate the MixCut on the manifests without recomputing all features again?

**Comment by pzelasko** [comment]
Yes, just run `mixed_cut = mixed_cut.truncate()` and it will read just the subset of computed features from disk. If you need to find the offset/duration of the speech cut in a mixed cut, you can find them in `mixed_cut.tracks` field (look for `.offset` and `.cut` attributes to find any metadata you need).
